indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -141,9 +141,25 @@ static RealtimeTuningConfig convertTuningConfig(ShardSpec spec, IndexTuningConfi
 
   private final ObjectMapper jsonMapper;
 
+  private String taskId;
+
+  /**
+   * This overloaded constructor without @JsonCreator exists to allow for
+   * unit tests simulating a call to IndexTask without taskResource.
+   */
+  public IndexTask(
+      String id,
+      IndexIngestionSpec ingestionSchema,
+      ObjectMapper jsonMapper
+  )
+  {
+    this(id, new TaskResource(makeId(id, ingestionSchema), 1), ingestionSchema, jsonMapper);
```

Comment:
Good point - it's much better to let the default action occur via the regular constructor.